_posts/2022-01-03-contributing-for-the-first-time-to-a-project.adoc

```diff
@@ -0,0 +1,174 @@
+
+- Add `.devfile.yaml` or `devfile.yaml` file to the root of the project in order to specify the relevant components and commands in a declarative way.
+
+The devfile provides easy to configure, highly reproducible definitions of portable developer environments.
+For more information about devfiles, see the link:https://devfile.io/[Devfile Documentation].
+
+NOTE: Omitting the creation of the devile file will result in using a default container based on the link:https://quay.io/repository/devfile/universal-developer-image[Universal Developer Image]
+which provides different languages (including Java, Node.js, Python, PHP, Golang, etc) and tools (including `curl`, `jq`, `yq`, etc).
+
+.The recommended file structure of the repository
+----
+  |--- devfile.yaml
+  |--- README.md
+  |___ .che
+         |___ che-editor.yaml
+         |___ che-theia-plugins.yaml
+  |___ .vscode
+         |___ extensions.json
+         |___ launch.json
+----
+
+- Add `.vscode/extensions.json` file to the root of the project in order to add extensions:
+
+[source,json]
+----
+{  
+  "recommendations": [
+    "redhat.java"  
+  ]  
+}
+----
+
+IMPORTANT: The list of the VS Code extension is available in the {prod} plug-in registry.
+If the specified extensions are not available in the registry, the workspace will start, but the extensions will not be added to a workspace. 
+
+- Optionally add `.vscode/launch.json` file for specifying `Run` and `Debug` commands:
+
+[source,json]
+----
+{
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "type": "java",
+            "name": "Debug (Attach) - Remote",
+            "request": "attach",
+            "hostName": "localhost",
+            "port": 5005
+        }
+    ]
+}
+----
+
+- Optionally add `.che/che-theia-plugins.yaml` file if extra customizations of extensions' containers are required on the {orch-name} level. For example, overriding the default memory limit of `redhat.java` sidecar container:
+
+[source,yaml]
+----
+- id: redhat/java  
+  override:  
+    sidecar:  
+      memoryLimit: 2Gi
+----
+
+- Optionally add `.che/che-editor.yaml` file if extra customizations of editors' containers are required on the {orch-name} level. For example, overriding the default memory limit of the `theia-ide` container:
+
+[source,yaml]
+----
+- id: eclipse/che-theia/latest          # mandatory
+  registryUrl: https://my-registry.com  # optional
+  override:                             # optional
+    containers:
+      - name: theia-ide
+        memoryLimit: 1280Mi
+----
+
+- Add the {prod} badge to the project `README.md` file with a link for starting a workspace:
+
+.The official {prod} badge
+image::https://www.eclipse.org/che/contribute.svg[Contribute Badge]
+{empty}
+
+====
+`[![Contribute](https://www.eclipse.org/che/contribute.svg)](pass:c,a,q[{prod-url}#https://github.com/my/repository])`
```

Comment:
I think it makes more sense to use [Che svg icon from simpleicons.org](https://simpleicons.org/?q=eclipse-che) and use shields.io to generate the "official" badge:

![](https://img.shields.io/static/v1?label=eclipse&message=che&logo=eclipseche)
![](https://img.shields.io/static/v1?label=code%20with&message=che&logo=eclipseche&color=FDB940&labelColor=525C86)
![](https://img.shields.io/static/v1?label=eclipse&message=che&logo=eclipseche&color=FDB940&labelColor=525C86&style=plastic)
![](https://img.shields.io/static/v1?label=eclipse&message=che&logo=eclipseche&color=FDB940&labelColor=525C86&style=flat-square)
![](https://img.shields.io/static/v1?label=eclipse&message=che&logo=eclipseche&style=social)


```
https://img.shields.io/static/v1?label=eclipse&message=che&logo=eclipseche
https://img.shields.io/static/v1?label=code%20with&message=che&logo=eclipseche&color=FDB940&labelColor=525C86
https://img.shields.io/static/v1?label=eclipse&message=che&logo=eclipseche&color=FDB940&labelColor=525C86&style=plastic
https://img.shields.io/static/v1?label=eclipse&message=che&logo=eclipseche&color=FDB940&labelColor=525C86&style=flat-square
https://img.shields.io/static/v1?label=eclipse&message=che&logo=eclipseche&style=social
```

Reply:
sounds good, I can add it as an alternative to https://www.eclipse.org/che/contribute.svg